### ISSUE TITLE
Remove references to cookbook@version style pinning

### ISSUE THREAD

**Issue by cheeseplus** [chef-boneyard/chef-web-docs-2016#574]
Remove any and all references to this pinning solution that has been long deprecated and unsupported.

`{"run_list":["recipe[nginx@1.2.3]"]}`

Souce: https://docs.chef.io/cookbook_versions.html#run-lists

**Comment by jamescott** [comment]
https://github.com/chef/chef-docs/commit/9a21498697ab73fffde3a190b53098cde3676f50

That gets it out of the root docs.chef.io pages and should get it out of the 12.x versioned docs. The 11.x docs will need to be rebuilt to verify, but that will be in the future (next time I rebuild those docs).

**Comment by Tech356** [comment]
@cheeseplus When was cookbook@version style pinning deprecated? I have been using this style extensively and I didn't know it wasn't supported.

**Comment by cheeseplus** [comment]
@Tech356 it's undocumented and something that predated the `cookbook_versions` key for Environments. So while technically it is "supported" from a code perspective it is not _encouraged_.

**Comment by Tech356** [comment]
@cheeseplus I would disagree with the 'undocumented' part, it was documented until this issue and related commit. I assume policy files are the encouraged way forward?

**Comment by cheeseplus** [comment]
An errant passing reference doesn't constitute "documented" which is the very reasons why I requested it's removal. As with lots of pieces of software it's an artifact of a time long past and not something new users should be consuming.

While Policyfiles can play a part here I'd say that Environment `cookbooks_versions` are the preferred mechanism for enforcing constraints with multiple cookbook versions in play. Policyfiles do overlap here but the constraints are enforced by a different (and new) mechanism.

To be clear the terminology in the initial report was inaccurate - the style is not deprecated nor unsupported from a code perspective. It's that the interactions of this style conflict with the current "best practices" and we don't want to confuse anybody by highlighting a vestigial and poorly documented "feature" that isn't intended to be the constraint mechanism.
